web/vite.config.ts

```diff
@@ -0,0 +1,79 @@
+import path from 'node:path'
+import { defineConfig } from 'vite'
+import react from '@vitejs/plugin-react'
+import { resolve } from 'path';
+import svgr from "vite-plugin-svgr";
+import istanbul from 'vite-plugin-istanbul';
+import { sentryVitePlugin } from "@sentry/vite-plugin";
+
+export default defineConfig({
+  build: {
+    sourcemap: true,
+    outDir: 'build',
+  },
+  optimizeDeps: {
+    include: ['@mui/material/Tooltip', '@emotion/styled/base', '@mui/material/Unstable_Grid2'],
+  },
+  plugins: [
+    {
+      name: 'build-html',
+      apply: 'build',
+      transformIndexHtml(html) {
+        return html.replace(
+          /<meta name="viewport" content="width=device-width, initial-scale=1.0">/,
+          `<meta name="viewport" content="width=device-width, initial-scale=1.0">
+         <script src="config.js"></script>
+    <script type="text/javascript">
+      window.env = config
+    </script>
+        `,
+        )
+      }
+    },
+    react({
+      jsxImportSource: "@emotion/react",
+      babel: {
+        plugins: ["@emotion/babel-plugin"],
+      },
+    }),
+    , svgr(),
+    istanbul({
+      include: 'src/*',
+      exclude: ['node_modules', 'test/'],
+      extension: ['.js', '.ts'],
+      cypress: true
+    }),
+    sentryVitePlugin({
+      org: "bcps-wps",
+      project: "frontend",
+      authToken: process.env.SENTRY_AUTH_TOKEN,
+      sourcemaps: {
+        filesToDeleteAfterUpload: ['**/*.map'],
+      },
+    }),
+  ],
+  server: {
+    port: Number(process.env.PORT) || 3000,
+    watch: {
+      ignored: [
```

Comment:
Should we also ignore `*.test.*` files?

Reply:
i added these for cypress tests running. When that is running these directories get changed and so dev server restarts. But the test files won't get changed so I think we're ok.

Reply:
Does this config only apply to cypress tests though? If a developer is running the vite server locally and making changes to component tests will this trigger dev server restarts?

Reply:
yea. is that a problem? it's like instantatenous. but i can add tests to ignored.

Reply:
Not sure if it's a problem, we could add that after if it becomes one.